src/visualization/GraphRenderer.cc

```diff
@@ -512,4 +512,53 @@ bool GraphRenderer::isPositionNode(sf::Vector2f pos, vector<DrawNode>& Nodes, ve
         }
     }
     return false;
-} //&& temp.x >= positionToCoords(rects.at(temp.y).at(0).getOrigin()).x && temp.x < positionToCoords(rects.at(temp.y).back().getOrigin()).x)
+}
+
+vector<sf::Vector2f> GraphRenderer::calcNewNodeCoord(Gamemaster& gamemaster, vector<Node>& nodeList){
+    // for testing   length of kmer = 5 sequence lenght = 10 number of sequences = 2 mutationprop. = 0 
+    sf::Vector2f t1(0, 1);
+    sf::Vector2f t2(2, 1);
+    sf::Vector2f t3(1, 2);
+    sf::Vector2f t4(2, 2);
+    vector<sf::Vector2f> new_coord = {t1, t2, t3, t4};
+    //TODO: we need a new coordinate calculation function
+    return new_coord;
+}
+
+void GraphRenderer::calcAnimationSpeed(unsigned int size){
+    animationSpeed.resize(size + 1);
+    for (unsigned int i = 0; i <= size; i++) {
+        double x = i / (double)size ;
+        //animationSpeed[i] = x;   // constant speed
+        //animationSpeed[i] = x * x; // quaternary acceleration
+        animationSpeed[i] = x * (3 + (x - 3) * x); // acceleration and slowing down
```

Comment:
I don't see the need for such a vector. I would have expected a formula for the speed that only depends on the time t the animation is running.